### ISSUE TITLE
[Bug] Very high CPU and memory usage

### ISSUE THREAD

**Issue by 010011x** [OpenBB-finance/OpenBBTerminal#5104]
**Describe the bug**
I install openbb via conda. Immediately after starting **openbb**, memory and CPU usage of the openbb process jump to very high numbers. 
Unfortunately, this makes this amazing software unusable for me.

**To Reproduce**
```
conda env create -n obb --file https://raw.githubusercontent.com/OpenBB-finance/OpenBBTerminal/main/build/conda/conda-3-9-env.yaml
conda activate obb
pip install openbb --no-cache-dir
openbb
```

**Screenshots**
**top**

![image](https://github.com/OpenBB-finance/OpenBBTerminal/assets/84412174/04a7bb3c-b034-4ca3-9229-8081361c22aa)
**perf**

![image](https://github.com/OpenBB-finance/OpenBBTerminal/assets/84412174/4635fd84-9899-4f0b-878d-0f1293ef2e34)

**Desktop (please complete the following information):**
LXQT

- OS: EndeavourOS
- Python version: 3.9.13

**Comment by 010011x** [comment]
strance output

```
recvmsg(10, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
clock_gettime(CLOCK_MONOTONIC, {tv_sec=13223, tv_nsec=41713144}) = 0
clock_gettime(CLOCK_MONOTONIC, {tv_sec=13223, tv_nsec=41760500}) = 0
clock_gettime(CLOCK_MONOTONIC, {tv_sec=13223, tv_nsec=41806390}) = 0
clock_gettime(CLOCK_MONOTONIC, {tv_sec=13223, tv_nsec=41852140}) = 0
clock_gettime(CLOCK_MONOTONIC, {tv_sec=13223, tv_nsec=41899915}) = 0

```

**Comment by deeleeramone** [comment]
Curious if your installation is getting configured as Linux/AMD instead AARCH.  It might be a good idea to try using Python 3.10, just replace the `9` in the URL.

**Comment by 010011x** [comment]
> Curious if your installation is getting configured as Linux/AMD instead AARCH. It might be a good idea to try using Python 3.10, just replace the `9` in the URL.

can you please elaborate ? I'm on AMD
```
uname -m
x86_64
```

Tried doing Python 3.10, same issue.

**Comment by deeleeramone** [comment]
I see, so much for that theory! The reason I was curious architecture emulation can be a real drag sometimes, but that does not fit with your symptoms.  It is extremely difficult to support all Linux configurations, because there are many variables and variations.

Some distributions work better with Miniforge: https://github.com/conda-forge/miniforge#miniforge

What is your CPU generation, make, and model?  Please be as specific as possible with your hardware and software configurations.  Otherwise, we don't really have a starting point. :)

**Comment by 010011x** [comment]
> Are you running 32-bit processes? What is your exact Linux distribution, build, and configuration?
> 
> Can this be a factor?
> 
> ![Screenshot 2023-06-06 at 10 03 36 AM](https://user-images.githubusercontent.com/85772166/243773243-9f92e537-7461-4e8f-a13c-6546e6bf6ad1.png)
> 
> https://endeavouros.com/latest-release/
> 
> ![Screenshot 2023-06-06 at 10 06 13 AM](https://user-images.githubusercontent.com/85772166/243773717-8031fd01-637e-4658-b8f2-ce12c3d2f2b8.png)

doesn't seems like any of this is related.
I'm just using stock endeavouros, 64bit, amd. 

so there is no way I can debug what thread is using all memory ? I've tried cProfile and stuff with no luck. There is nothing that comes up to your mind in regards to cpu consumption ? 

Also tried installing `poetry install` without `poetry.lock`.

**Comment by 010011x** [comment]
As a Python developer myself, I have a bunch of python projects and libs, browsers, editors etc etc.

none of the other Python processing are using anywhere close of this amount of CPU. 
I have other conda environments and all is good. 

I've been going thought openbb source code, from entry point and to input loops trying to find where bottleneck might be. I'm very sure it's something in the code, but you never know.

**Comment by 010011x** [comment]
I'm sure others might have similar issue but just didn't noticed. It's more noticeable on my machine.

Docker way works fine, but it's too much overhead for me.

Love the idea of this project, but will revisit in next year when it's more usable.

Thanks a lot for your help.

**Comment by deeleeramone** [comment]
> Love the idea of this project, but will revisit in next year when it's more stable / usable. This is clearly a no-go.

You are currently the only user reporting this issue, so it is likely something related to your local system configuration.  As you showed, you are running Intel, `x86/64`, processes instead of `ARM/AMD`, which specifically supports AMD processors.  I would look into that.

**Comment by 010011x** [comment]
> > Love the idea of this project, but will revisit in next year when it's more stable / usable. This is clearly a no-go.
> 
> You are currently the only user reporting this issue, so it is likely something related to your local system configuration. As you showed, you are running Intel, `x86/64`, processes instead of `ARM/AMD`, which specifically supports AMD processors. I would look into that.

My processor is neither Intel or ARM.

https://en.wikichip.org/wiki/amd/ryzen_7/pro_4750u

`ISA | x86-64 (x86)`

**Comment by deeleeramone** [comment]
@hjoaquim, have you ran into this before?

**Comment by hjoaquim** [comment]
It looks like I have the same issue (if I'm interpreting the `htop` correctly):
![image](https://github.com/OpenBB-finance/OpenBBTerminal/assets/48914296/e2f2d5ec-8a63-4994-baf9-92f132150cec)

Although the CPU goes to 100% I don't really experience any performance issues on my machine.

**Comment by 010011x** [comment]
> It looks like I have the same issue (if I'm interpreting the `htop` correctly): ![image](https://user-images.githubusercontent.com/48914296/243980991-e2f2d5ec-8a63-4994-baf9-92f132150cec.png)
> 
> Although the CPU goes to 100% I don't really experience any performance issues on my machine.

Thank you so much for reproducing and confirming. 
I will try to debug this.

**Comment by 010011x** [comment]
Created follow up https://github.com/OpenBB-finance/pywry/issues/77

I think we are close

**Comment by tehcoderer** [comment]
Replied in PyWry issue thread! 😊

Fixed in [#78](https://github.com/OpenBB-finance/pywry/pull/78) `v0.5.12` !
